src/Tasks/ResolveAssemblyReferences/Services/ResolveAssemblyReferenceComparer.cs

```diff
@@ -0,0 +1,187 @@
+
+            return fieldsEqual &&
+                   AreStringListsEqual(x.AllowedAssemblyExtensions, y.AllowedAssemblyExtensions) &&
+                   AreStringListsEqual(x.AllowedRelatedFileExtensions, y.AllowedRelatedFileExtensions) &&
+                   AreTaskItemListsEqual(x.Assemblies, y.Assemblies) &&
+                   AreTaskItemListsEqual(x.AssemblyFiles, y.AssemblyFiles) &&
+                   AreStringListsEqual(x.CandidateAssemblyFiles, y.CandidateAssemblyFiles) &&
+                   AreTaskItemListsEqual(x.FullFrameworkAssemblyTables, y.FullFrameworkAssemblyTables) &&
+                   AreStringListsEqual(x.FullFrameworkFolders, y.FullFrameworkFolders) &&
+                   AreStringListsEqual(x.FullTargetFrameworkSubsetNames, y.FullTargetFrameworkSubsetNames) &&
+                   AreTaskItemListsEqual(x.InstalledAssemblySubsetTables, y.InstalledAssemblySubsetTables) &&
+                   AreTaskItemListsEqual(x.InstalledAssemblyTables, y.InstalledAssemblyTables) &&
+                   AreStringListsEqual(x.LatestTargetFrameworkDirectories, y.LatestTargetFrameworkDirectories) &&
+                   AreTaskItemListsEqual(x.ResolvedSDKReferences, y.ResolvedSDKReferences) &&
+                   AreStringListsEqual(x.SearchPaths, y.SearchPaths) &&
+                   AreStringListsEqual(x.TargetFrameworkDirectories, y.TargetFrameworkDirectories) &&
+                   AreStringListsEqual(x.TargetFrameworkSubsets, y.TargetFrameworkSubsets);
+        }
+
+        internal static bool CompareOutput(ResolveAssemblyReferenceResponse x, ResolveAssemblyReferenceResponse y)
+        {
+            if (x == y)
+            {
+                return true;
+            }
+
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            return x.DependsOnNETStandard == y.DependsOnNETStandard &&
+                   x.DependsOnSystemRuntime == y.DependsOnSystemRuntime &&
+                   AreTaskItemListsEqual(x.CopyLocalFiles, y.CopyLocalFiles) &&
+                   AreTaskItemListsEqual(x.FilesWritten, y.FilesWritten) &&
+                   AreTaskItemListsEqual(x.RelatedFiles, y.RelatedFiles) &&
+                   AreTaskItemListsEqual(x.ResolvedDependencyFiles, y.ResolvedDependencyFiles) &&
+                   AreTaskItemListsEqual(x.ResolvedFiles, y.ResolvedFiles) &&
+                   AreTaskItemListsEqual(x.SatelliteFiles, y.SatelliteFiles) &&
+                   AreTaskItemListsEqual(x.ScatterFiles, y.ScatterFiles) &&
+                   AreTaskItemListsEqual(x.SerializationAssemblyFiles, y.SerializationAssemblyFiles) &&
+                   AreTaskItemListsEqual(x.SuggestedRedirects, y.SuggestedRedirects); 
+            //&&
+            //       AreTaskItemListsEqual(x.Assemblies, y.Assemblies) &&
+            //       AreTaskItemListsEqual(x.AssemblyFiles, y.AssemblyFiles);
+        }
+
+        private static bool AreStringListsEqual(string[] x, string[] y)
```

Comment:
Have you considered using `Enumerable.SequenceEqual()`?

Reply:
In my first version I used it, but that required strings array to be not null, so I decided to go with this approach.